### PR TITLE
fixes issue where config.ext was not set (FCREPO-2347)

### PR DESCRIPTION
**Pull request by jwestgard**
Import-export no longer supports option of setting ext explicitly, but infers from rdfLang the proper extension.  The verification tool must make this same inference.

The tool still defaults to turtle if the lang is not set, but if it is set, it checks the value against a mapping dictionary. If lang is set but the value is not found in the dict, the tool reports the error and breaks off.

Fixes https://jira.duraspace.org/browse/FCREPO-2347

### CODE REVIEW

**Comment by whikloj** [comment]
@jwestgard I can modify the import-export-script to add the extension it is using to the config file if that would avoid you having to store a second mapping.

**Comment by jwestgard** [comment]
@whikloj, thanks, in many ways that would be the simpler solution, but on the other hand I wonder whether that would confuse users of import export who were editing their config manually -- that is, since the tool effectively would not respect what was in there for the extension.  Anyway, might be something to discuss with the stakeholders, and this PR lets the verification tool work in the meantime.

**Comment by bseeger** [comment]
I think it might be best to duplicate the map here and not try to put it in the config, for the reason that Josh specifies.    Makes me wonder if there's an external ontology/something out there that both code bases could pull from.

**Comment by jwestgard** [comment]
@bseeger, I gather that the import export tool is using some resource provided by the Apache Jena project. I'm not sure if there's a way to make use of the same in Python, but would be good to look into. Any chance you would have time to look into it?

**Comment by bseeger** [comment]
@jwestgard - sure I'll take a quick look

**Comment by whikloj** [comment]
@jwestgard @bseeger Perhaps instead of the config we should be adding this information to the audit log. So you can perform various exports and the audit log holds what your configuration was at that point in time. Especially because if you have a config file and use command line arguments, then the config file is not actually what you ran.

**Comment by bseeger** [comment]
@whiklog, @jwestgard  - I really like this idea, Jared.  We could grab some info from that file, once the format is set.  The one snag I can think of is whether or not the audit log will be available for import/export testing... if it is available, that actually provides a different way to look at verifying things, with the use of what's in that file.  We would then know exactly what to expect and not have to worry about missing Resources, since we could verify against the audit log and use it sort of as a check list.   But that last part is outside the scope of this PR.

**Comment by whikloj** [comment]
@jwestgard @bseeger Wednesday I think. I can cram this information in the audit log anytime, we'd just need to agree on a format. I'd probably suggest a json object as the header. So you can grab the first line decode it to get the configuration, then you could iterate over the rest of the lines to check objects. Currently the last line is a count or an error if it failed.

**Comment by jwestgard** [comment]
I think the only thing scheduled is the retrospective on Friday 2pm EDT.

**Comment by whikloj** [comment]
@jwestgard I think we have a checkin call on Wednesday and the sprint debrief on Friday. Both at noon ET.

File: verify.py [hunk 1]
```diff
@@ -108,10 +118,20 @@ def __init__(self, configfile, auth, logger):
...
                 self.lang = value
 
-        # if not specified in config, set default ext & lang to turtle
+        # if lang not specified in config, set the ext & lang to turtle
         if not hasattr(self, 'lang'):
             self.ext = '.ttl'
             self.lang = 'text/turtle'
+        # set the ext based on the rdfLang
+        else:
+            if self.lang in EXT_MAP:
```
Comment (bseeger) [inline verify.py]: Same change of `lang` to rdfLang

File: verify.py [hunk 2]
```diff
@@ -108,10 +118,20 @@ def __init__(self, configfile, auth, logger):
...
+                self.ext = EXT_MAP[self.lang]
```
Comment (bseeger) [inline verify.py]: Another `lang`

File: verify.py [hunk 3]
```diff
@@ -108,10 +118,20 @@ def __init__(self, configfile, auth, logger):
             elif key == "rdfLang":
                 self.lang = value
 
-        # if not specified in config, set default ext & lang to turtle
+        # if lang not specified in config, set the ext & lang to turtle
         if not hasattr(self, 'lang'):
```
Comment (bseeger) [inline verify.py]: (Sorry for any confusion in my comments... I think I'm less confused now...)  `'lang'` should be `'rdfLang'`

Comment (bseeger) [inline verify.py]: Should we fix this here or in a separate PR?

**Comment by jwestgard** [comment]
@bseeger : for whatever reason I cannot locate your comments regarding rdfLang on Github, but I did get email notification. For what it's worth, I did not strive to align the names of the config properties with the keys in the config file itself, but it would perhaps be a good idea to do so.  I would say that ought to be its own ticket/PR, though, since it should be done systematically for the whole script and goes beyond the scope of this PR.

**Comment by bseeger** [comment]
@jwestgard  - I deleted my comments when I realized that I was a little confused about something. You can completely ignore those emails.   I don't feel the need to align the var names to the config names.

File: verify.py [hunk 4]
```diff
@@ -108,10 +118,20 @@ def __init__(self, configfile, auth, logger):
...
             self.lang = 'text/turtle'
+        # set the ext based on the rdfLang
+        else:
+            if self.lang in EXT_MAP:
+                self.ext = EXT_MAP[self.lang]
+            else:
+                logger.error(
+                    'Unrecognized rdf dialect specified in config file!'
+                    )
+                print('Unrecognized rdf dialect specified in config file!')
```
Comment (jwestgard) [inline verify.py]: What would be the protocol here -- I'm assuming I make the edits? Can I do that directly in Github?

**Pull request merged**